### ISSUE TITLE
Unable to connect to cloud sql with node Appengine and TypeOrm

### ISSUE THREAD

**Issue by nishantkCB** [typeorm/typeorm#3263]
**Issue type:**

[X] question
[ ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[X] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[X ] `0.2.7` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I am using typeorm and with extra.socketPath in ormconfig file .

Error : Error: connect ENOENT /cloudsql/[project:region:instance] at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1113:14)

ormConfig.json file contains below json . { "type": "mysql", "extra": { "socketPath": "/cloudsql/[project:region:instance]" }, "username": "root", "password": "xxxx", "database": "yyy", "synchronize": true, "logging": false, "entities": [ "build/entity/**/*.js" ] }

**Comment by nishantkCB** [comment]
> Are you using `/cloudsql/[project:region:instance]` as value or this is just placeholder to protect information about your project, region and instance in gcloud?

its just a place holder . Any way this is resolved i have used cloud sql ip as host and create a firewall rule to allow it which is working .

**Comment by vlapo** [comment]
Glad to hear it. We are using typeorm with pgsql on appengine and do not have any problem with connect over socket.

**Comment by etudor** [comment]
Is there any fix for this issue other than using cloud SQL IP?
My config
```
{ type: 'mysql',
  port: '3306',
  username: 'myuser',
  password: 'mypass',
  database: 'mydb',
  synchronize: false,
  logging: true,
  entities: [ 'src/entity/**/*.ts' ],
  migrations: [ 'src/migration/**/*.ts' ],
  subscribers: [ 'src/subscriber/**/*.ts' ],
  extra: { socketPath: '/cloudsql/xxx:xx:xx' } }
```

**Comment by VaivanOmmeren** [comment]
I fixed it without having to add cloud SQL IP.

You need to also define `/cloudsql/[project:region:instance]` as your host and not only as socketPath.
If no host is specified it will automatically try to connect to `127.0.0.1`

Hope that helps anyone with their issue.

**Comment by jeromeSH26** [comment]
hi guys,

although I followed your configurations, I have a similar issue. Here is my configration file :

```
let confBase: PostgresConnectionOptions = {
	type: "postgres",
	host: dbHOST as string,
	port: parseInt(dbPORT as string, 10),
	username: dbUSER,
	password: dbPWD,
	extra: { socketPath: gcpcloud },
	database: dbDATABASE,
	synchronize: true,
	logging,
};
```

where both `dbHOST `and` gcpcould` are string like this `"/cloudsql/THEPROJECT:THEREGION:THEINSTANCE"`

what am I missing here ?

The connection from a GCLOUD terminal using psql is working fine, so the database is UP and the user/pwd are OK.

Thks

**Comment by VaivanOmmeren** [comment]
This is an example of my configuration

```typescript
type: "postgres",
    host: /cloudsql/project:region:instance,
    extra: {
         socketPath: /cloudsql/project:region:instance
    },
    username: username,
    password: password,
    database: database,
    entities: [
        "dist/models/*.js"
    ],
    synchronize: true,
    logging: false
```

Port doesn't need to be specified when connecting through socket, also you seem to be missing your entity reference.

**Comment by jeromeSH26** [comment]
hi thks for your reply
I found the issue, the host has to be the PRIVATE IP of the VM where.the DB is hosted, and not either the public IP, nore the connection similair to the socketPath.
Th weird thing is that in your configuration HOST and SocketPatch are both the same and it is working, whereas it failed in mine...

```
let confBase: PostgresConnectionOptions = {
	type: "postgres",
	host: dbPRIVATE_IP,
	port: parseInt(dbPORT as string, 10),
	username: dbUSER,
	password: dbPWD,
	extra: { socketPath: gcpcloud },
	database: dbDATABASE,
	synchronize: true,
	logging,
};
```

Nb :  for the entities, they are added to the basic options later in the code ;-)

**Comment by samuliasmala** [comment]
I tried to connect to Google Cloud SQL (mysql instance) when using TypeORM in App Engine Flexible. I used the tcp port approach so I had this in my `app.yaml` (<INSTANCE_CONNECTION_NAME> of course replaced with the actuan value):

```yaml
beta_settings:
  cloud_sql_instances: <INSTANCE_CONNECTION_NAME>=tcp:3306
```

and connection options for TypeORM 
``` js
{
  type: 'mysql',
  host: '172.17.0.1',
  username: process.env.DB_USER,
  password: process.env.DB_PASSWORD,
  database: process.env.DB_DATABASE,
  synchronize: false,
  port: 3306,
}
```

I used the same app.yaml, host, port, username etc. values and got the connection in [example project ](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/tree/master/cloud-sql/mysql/mysql )working. Any ideas what could be the problem? Could it be the database driver? The example project is using `promise-mysql`, is the TypeORM database different and somehow uncompatible with Google Cloud SQL?

Btw I got this working by opening the database to public internet and replacing the host with the public ip. But I would rather not open it.
